crates/chia-bls/src/cached_bls.rs

```diff
@@ -43,15 +44,16 @@ impl BLSCache {
         Self { cache }
     }
 
-    pub fn aggregate_verify<M: AsRef<[u8]>>(
+    pub fn aggregate_verify<M: Iterator<Item = T>, T: AsRef<[u8]>, P: Iterator<Item = U>, U: Borrow<PublicKey>>(
```

Comment:
Wouldn't IntoIterator be more versatile? You could pass in an actual vector, slice, or array, without manually creating an iterator over it?

Existing iterators also implement IntoIterator for themselves I believe

Reply:
Ah, fantastic. This is very useful. I did feel like my solution was not optimal and this is the exact thing I was looking for thank you